packages/react/src/components/Dropdown/utilities/DropdownSizePosCache.ts

```diff
@@ -14,24 +14,32 @@ import type { IDropdownOption } from '../Dropdown.types';
 export class DropdownSizePosCache {
   private _cachedOptions: IDropdownOption[];
   private _displayOnlyOptionsCache: number[];
```

Comment:
- _displayOnlyOptionsCache tracks the items that are not focusable: headers and dividers
- _notSelectableOptionsCache caches all non-selectable option: headers and dividers but also hidden options.